cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -710,9 +632,9 @@ private EntityId toEntityId(TAuthorizable tAuthorizable, @Nullable EntityId pare
       case ARTIFACT:
         Artifact artifact = (Artifact) authorizable;
         assertNonNullParent(parent, Authorizable.AuthorizableType.ARTIFACT);
-        // TODO: this method will be remmoved CDAP-12100
-        //noinspection ConstantConditions
-        return ((NamespaceId) parent).artifact(artifact.getArtifactName(), artifact.getArtifactVersion());
+        // TODO: Ideally we should be constructing cdap authorizable from the tAuthorizable. CDAP authorizable does
```

Comment:
Instead it would be more useful to say why it is okay to construct an incorrect entity here.